src/app/tests/TestTimeSyncDataProvider.cpp

```diff
@@ -201,98 +207,65 @@ void TestDSTOffset(nlTestSuite * inSuite, void * inContext)
 
         dstObj.dstOffsetList = dstObj.dstOffsetList.SubSpan(1);
     }
 
     if (!dstObj.dstOffsetList.empty())
     {
         auto & dst = dstObj.dstOffsetList.data()[0];
-        NL_TEST_ASSERT(inSuite, dst.offset == 2);
-        NL_TEST_ASSERT(inSuite, dst.validStarting == 2);
-        NL_TEST_ASSERT(inSuite, !dst.validUntil.IsNull());
-        NL_TEST_ASSERT(inSuite, dst.validUntil.Value() == 3);
+        EXPECT_EQ(dst.offset, 2);
+        EXPECT_EQ(dst.validStarting, 2u);
+        EXPECT_FALSE(dst.validUntil.IsNull());
+        EXPECT_EQ(dst.validUntil.Value(), 3u);
 
         dstObj.dstOffsetList = dstObj.dstOffsetList.SubSpan(1);
     }
 
     if (!dstObj.dstOffsetList.empty())
     {
         auto & dst = dstObj.dstOffsetList.data()[0];
-        NL_TEST_ASSERT(inSuite, dst.offset == 3);
-        NL_TEST_ASSERT(inSuite, dst.validStarting == 3);
-        NL_TEST_ASSERT(inSuite, dst.validUntil.IsNull());
+        EXPECT_EQ(dst.offset, 3);
+        EXPECT_EQ(dst.validStarting, 3u);
+        EXPECT_TRUE(dst.validUntil.IsNull());
 
         dstObj.dstOffsetList = dstObj.dstOffsetList.SubSpan(1);
     }
 
-    NL_TEST_ASSERT(inSuite, dstObj.dstOffsetList.empty());
+    EXPECT_TRUE(dstObj.dstOffsetList.empty());
 }
 
-void TestDSTOffsetEmpty(nlTestSuite * inSuite, void * inContext)
+TEST_F(TestTimeSyncDataProvider, TestDSTOffsetEmpty)
 {
     TestPersistentStorageDelegate persistentStorage;
     TimeSyncDataProvider timeSyncDataProv;
     timeSyncDataProv.Init(persistentStorage);
 
     TimeSyncDataProvider::DSTOffsetObj dstObj;
 
-    NL_TEST_ASSERT(inSuite, CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND == timeSyncDataProv.LoadDSTOffset(dstObj));
-    NL_TEST_ASSERT(inSuite, !dstObj.dstOffsetList.begin());
-    NL_TEST_ASSERT(inSuite, dstObj.validSize == 0);
-}
-
-const nlTest sTests[] = { NL_TEST_DEF("Test TrustedTimeSource store load", TestTrustedTimeSourceStoreLoad),
-                          NL_TEST_DEF("Test TrustedTimeSource empty", TestTrustedTimeSourceEmpty),
-                          NL_TEST_DEF("Test default NTP store load", TestDefaultNTPStoreLoad),
-                          NL_TEST_DEF("Test default NTP empty", TestDefaultNTPEmpty),
-                          NL_TEST_DEF("Test time zone store load", TestTimeZoneStoreLoad),
-                          NL_TEST_DEF("Test time zone (empty list)", TestTimeZoneEmpty),
-                          NL_TEST_DEF("Test DSTOffset", TestDSTOffset),
-                          NL_TEST_DEF("Test DSTOffset (empty list)", TestDSTOffsetEmpty),
-                          NL_TEST_SENTINEL() };
-
-int TestSetup(void * inContext)
-{
-    VerifyOrReturnError(CHIP_NO_ERROR == chip::Platform::MemoryInit(), FAILURE);
-    return SUCCESS;
-}
-
-int TestTearDown(void * inContext)
-{
-    chip::Platform::MemoryShutdown();
-    return SUCCESS;
+    EXPECT_EQ(CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND, timeSyncDataProv.LoadDSTOffset(dstObj));
+    EXPECT_FALSE(dstObj.dstOffsetList.begin());
```

Comment:
this reads awakward. Does `EXPECT_TRUE(dstObj.dstOffsetList.empty())` work ? Or maybe:

`EXPECT_EQ(dstObj.dstOffsetList.begin(), dstObj.dstOffsetList.end())`

or is this really wanting to check for nullability? maybe then we should compare with nullptr, however a compare by converting an iterator to a bool feels off.

Reply:
yes emtpy works fine, done!